apps/web/src/app/[locale]/challenge/[slug]/_context.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+import { createContext, useState, useContext, type PropsWithChildren } from 'react';
+
+interface TrackVisibilityContextType {
+  isTrackTitleVisible: boolean | null;
+  setIsTrackTitleVisible: React.Dispatch<React.SetStateAction<boolean | null>>;
+}
+
+export const TrackVisibilityContext = createContext<TrackVisibilityContextType | undefined>(
+  undefined,
+);
+
+export const TrackVisibiltyProvider = ({ children }: PropsWithChildren) => {
+  const [isTrackTitleVisible, setIsTrackTitleVisible] = useState<boolean | null>(null);
+
+  return (
+    <TrackVisibilityContext.Provider value={{ isTrackTitleVisible, setIsTrackTitleVisible }}>
+      {children}
+    </TrackVisibilityContext.Provider>
+  );
+};
+
+export const useTrackVisiblity = () => {
```

Comment:
suggestion:
I think this name would be more informative. WDYT? Same for context, provider and state names
```suggestion
export const useTrackNavigationVisiblity = () => {
```

Reply:
W naming